go-controller/pkg/ovn/logical_switch_manager/logical_switch_manager.go

```diff
@@ -630,3 +629,17 @@ func NewL2IPAMAllocator(cidr *net.IPNet) (ipam.Interface, error) {
 	}
 	return subnetRange, nil
 }
+
+// GenerateRandMAC generates a random unicast and locally administered MAC address.
+// LOOTED FROM https://github.com/cilium/cilium/blob/v1.12.6/pkg/mac/mac.go#L106
+func GenerateRandMAC() (net.HardwareAddr, error) {
```

Comment:
we should probably pass this thing the logical switch, and assure the MAC is not present in any of its ports.

Can we do that in a follow-up ?

Reply:
hmm yeah, we should probably generate macs a little more intelligently here and ensure that mac is not already used somewhere else. Today we have the convenience of translating IPs to macs, and we allocate ips so we know it is unique.